mm2src/coins/z_coin.rs

```diff
@@ -776,6 +776,8 @@ pub struct ZcoinActivationParams {
     pub scan_blocks_per_iteration: u32,
     #[serde(default)]
     pub scan_interval_ms: u64,
+    #[serde(default)]
+    pub path_to_address: StandardHDCoinAddress,
```

Comment:
As only `account` field is used by Zcoin, `StandardHDCoinAddress` might be confusing if the user sets `is_change` and/or `address_index` and get the same resulting activated address. 
Would it be better to just use `account: u32` here?

Reply:
> As only account field is used by Zcoin

According to this https://zips.z.cash/zip-0032#sapling-key-path address indexes can be supported too, and depending on using `account` or `address_index` as the penultimate level there will be multiple payment addresses on the next/last level corresponding to different diversifiers. But I think we should keep it simple and use account as the penultimate level, but do you think we should use diversifiers to have multiple addresses? In this case `account: u32` will not be enough and we should have a `diversifier` field too.

> Would it be better to just use account: u32 here?

I can do that for now until I look more into this.

Reply:
> do you think we should use diversifiers to have multiple addresses?

As per ZIP32, `A group of such addresses shares the same full viewing key and incoming viewing key,`, so I assume our implementation should already work (detect unspent notes, etc.) for diversified addresses that were generated for the *given account*. So additional `diversifier` field should not be needed (worth testing it just in case).

 Another question is whether we would like to implement such diversified addresses generation
 > They MAY also support generating a stream of payment addresses for a given account, if they wish to maintain the user experience of giving a unique address to each recipient.
 
I assume Ywallet uses it in advanced mode via `New Snap Address` button - it will generate a new address that will be shown for a few seconds.
<img src="https://github.com/KomodoPlatform/komodo-defi-framework/assets/15745003/1f7d7bfc-a36e-4273-9d04-9d2a01af6daa" width=200 align="left">
<img src="https://github.com/KomodoPlatform/komodo-defi-framework/assets/15745003/298fed46-fe27-4965-8917-3d0ee2517aef" width=200>

> address indexes can be supported too

According to ZIP, we *may* support it too, and if we prefer to do so, I would propose to implement such derivation right away to avoid confusion caused by unused `address_index` field.

Also, change does not exist for shielded address, so `StandardHDCoinAddress` doesn't seem a fit here anyway because it still has `is_change` that won't be ever used.

Reply:
Added `account` field to `ZcoinActivationParams` and used it instead. Also added this conversation to the checklist https://github.com/KomodoPlatform/komodo-defi-framework/issues/1838#issuecomment-1675467570 to remember about looking more into how we want to handle `diversifier` and `address_index`.